internals/src/macros.rs

```diff
@@ -0,0 +1,63 @@
+//! Various macros used by the Rust Bitcoin ecosystem.
+//!
+
+/// Implements standard array methods for a given wrapper type
+#[macro_export]
+macro_rules! impl_array_newtype {
+    ($thing:ident, $ty:ty, $len:literal) => {
+        impl $thing {
+            /// Converts the object to a raw pointer
+            #[inline]
+            pub fn as_ptr(&self) -> *const $ty {
+                let &$thing(ref dat) = self;
```

Comment:
If we decide to guarantee validity `let inner: &[$ty; $len] = &self.0; inner as *const $ty` would be safer.

Reply:
I agree your code is more obviously correct, but is the existing code wrong?

Reply:
The macro doesn't define the type so supplying a wrong value to the macro could make this unsound. Although it's caught by `to_bytes()` method which wouldn't type check. I'd rather not rely on things far a way from this code - the main benefit of `unsafe` is being able to limit the scope of affected code.